### PR TITLE
Make settlement script look at everything owed 

### PR DESCRIPTION
**Pull request by znarf**
and not just current month.

Fix https://github.com/opencollective/opencollective/issues/4520

### CODE REVIEW

File: server/lib/host-metrics.js [hunk 1]
```diff
@@ -414,9 +414,9 @@ export async function getPendingHostFeeShare(host, { startDate, endDate, collect
           AND t."kind" = 'HOST_FEE_SHARE_DEBT'
           AND t."deletedAt" IS NULL
           AND ts."deletedAt" IS NULL
-          AND ts."status" IN ('OWED', 'INVOICED')
-          AND t."createdAt" >= :startDate
-          AND t."createdAt" <= :endDate
+          AND ts."status" IN ('OWED')
```
Comment (Betree) [inline server/lib/host-metrics.js:417]: Not sure about this one. If it's `INVOICED`, then it's still pending no?

File: cron/monthly/host-settlement.ts [hunk 2]
```diff
@@ -90,11 +90,10 @@ export async function run(baseDate: Date | moment.Moment = defaultDate): Promise
 FROM "Transactions" as t
 INNER JOIN "TransactionSettlements" ts ON ts."TransactionGroup" = t."TransactionGroup" AND t.kind = ts.kind
 WHERE t."CollectiveId" = :CollectiveId
-AND t."createdAt" >= :startDate AND t."createdAt" < :endDate
 AND t."kind" IN ('PLATFORM_TIP_DEBT', 'HOST_FEE_SHARE_DEBT')
 AND t."isDebt" IS TRUE
 AND t."deletedAt" IS NULL
-AND ts."status" != 'SETTLED'`,
+AND ts."status" = 'OWED'`,
```
Comment (Betree) [inline cron/monthly/host-settlement.ts:96]: Good change, this one could have resulted in duplicate settlements for the same transaction

File: server/lib/host-metrics.js (hunk 1 above)
Comment (znarf) [reply server/lib/host-metrics.js:417]: It depend the definition. In the context of the settlement process, we don't want to Invoice multiple time.

I think it's fine to move to this definition.

Comment (znarf) [reply server/lib/host-metrics.js:417]: Alternative is to pass an explicit `kind` when we want it.

Comment (Betree) [reply server/lib/host-metrics.js:417]: I agree for the settlement script. For host metrics however, when I look at my pending host fee share, I expect to see everything that I still haven't paid - invoiced or not.

Comment (znarf) [reply server/lib/host-metrics.js]: You might be right theoretically.

However, as far as I know, we're not surfacing this metric anymore, it's only used in the context of the settlement.

Still, I took the feedback in consideration and refactored, so there is no surprise.

**Pull request merged**